src/providers/portion-provider.ts

```diff
@@ -79,61 +98,57 @@ export class PortionProvider implements IPortionProvider {
-    portionAmount: CurrencyAmount
+    portionQuoteAmount: CurrencyAmount
   ): CurrencyAmount {
     switch (tradeType) {
       case TradeType.EXACT_INPUT:
-        return quoteGasAdjusted.subtract(portionAmount);
+        return quote;
       case TradeType.EXACT_OUTPUT:
-        return this.getQuoteGasAndPortionAdjustedForExactOut(
-          portionAmount,
-          quote,
-          quoteGasAdjusted,
-          amount
-        );
+        return quote.subtract(portionQuoteAmount);
       default:
         throw new Error(`Unknown trade type ${tradeType}`);
     }
   }
 
-  /**
-   * For exact out, there is no way we know the accurate quote gas and portion adjusted amount,
-   * because we are exact out quoting using the user requested amount.
-   *
-   * This method is a simple approximation that assumes the portion amount is a portion of the user requested amount.
-   * Then the portion amount can be adjusted to the quote gas adjusted amount, so that we can get a quote gas and portion adjusted.
-   *
-   * @param portionAmountToken this is the token out portion amount in decimal-less/unit-less token quantities
-   * @param quoteCurrencyAmount this is the token in quote amount in decimal-less/unit-less token quantities
-   * @param quoteGasAdjustedCurrencyAmount this is the token in quote adjusted gas amount in decimal-less/unit-less token quantities
-   * @param amount this is the token out amount requested by the swapper for exact out swap
-   * @private
-   */
-  private getQuoteGasAndPortionAdjustedForExactOut(
-    portionAmountToken: CurrencyAmount,
-    quoteCurrencyAmount: CurrencyAmount,
-    quoteGasAdjustedCurrencyAmount: CurrencyAmount,
-    amount: CurrencyAmount
+  getQuoteGasAdjusted(
+    tradeType: TradeType,
+    quoteGasAdjusted: CurrencyAmount,
+    portionQuoteAmount: CurrencyAmount
   ): CurrencyAmount {
-    // we have a simple heuristic to estimate the quote gas and portion adjusted for exact out
+    switch (tradeType) {
+      case TradeType.EXACT_INPUT:
+        return quoteGasAdjusted;
+      case TradeType.EXACT_OUTPUT:
+        return quoteGasAdjusted.subtract(portionQuoteAmount);
+      default:
+        throw new Error(`Unknown trade type ${tradeType}`);
+    }
+  }
 
-    const portionQuoteAmountToken = this.getPortionQuoteAmount(portionAmountToken, quoteCurrencyAmount, amount);
-    // 4. finally we have the estimated portion quote amount, we can add it to the quote gas adjusted amount,
-    //    i.e. quoteGasAdjustedCurrencyAmount = quoteGasAdjustedCurrencyAmount + portionQuoteAmountToken
-    return quoteGasAdjustedCurrencyAmount.add(portionQuoteAmountToken);
+  getQuoteGasAndPortionAdjusted(
+    tradeType: TradeType,
+    quoteGasAdjusted: CurrencyAmount,
+    portionAmount: CurrencyAmount
+  ): CurrencyAmount {
+    switch (tradeType) {
+      case TradeType.EXACT_INPUT:
+        return quoteGasAdjusted.subtract(portionAmount);
+      case TradeType.EXACT_OUTPUT:
+        return quoteGasAdjusted;
```

Comment:
so now this `quoteGasAdjusted` is actually `quoteGasAndPortionAdjusted` right? this is also kind of a misnomer at this point? Anything we can do about this?

Reply:
maybe call the function already with `quoteGasAdjusted` - `portionQuoteAmount` or somethinig?

Reply:
litearlly on the definition you say 

```
For exact out, quote gas adjusted amount is the best swap quote gas adjusted amount minus the portion quote token amount.
```

but actually you dont pass in that right now